File: Jeopardy/Jeopardy/ClientApp/src/Models/question.ts
export enum QuestionType
{
    Radio,
    MultipleChoice,
    Textfield,
    Slider,
    OrderedList
}
export interface Question {
    questionType: QuestionType;
    questionId: string;
    questionText: string;
    alreadyAnswered: boolean;
    difficulty: number;
    explanation: string;
}

export interface Answer {
    answerId: string;
    textAnswer: string;
}

export interface MultipleChoiceQuestion extends Question {
    possibleAnswers: Answer[];
    correctAnswerIds: string[];
}

export interface OrderedListQuestion extends Question {
    possibleAnswers: Answer[];
    correctOrderIds: string[];
}

export interface TextfieldQuestion extends Question {
    expectedAnswer: string[];
}

export interface RadioQuestion extends Question {
    possibleAnswers: Answer[];
    correctAnswerId: string;
}

export interface SliderQuestion extends Question {
    beginAnswer: number;
    endAnswer: number;
    step: number;
    correctAnswer: number;
    acceptedError: number;
}

File: Jeopardy/Jeopardy/Models/Questions/Question.cs
using Newtonsoft.Json;

namespace Jeopardy.Models
{
    public enum QuestionType
    {
        Radio,
        MultipleChoice,
        Textfield,
        Slider,
        OrderedList
    }

    public abstract class Question
    {
        [JsonProperty("questionType")]
        public QuestionType QuestionType { get; set; }

        [JsonProperty("questionId")]
        public string QuestionId { get; set; }

        [JsonProperty("questionText")]
        public string QuestionText { get; set; }

        [JsonProperty("alreadyAnswered")]
        public bool AlreadyAnswered { get; set; }

        [JsonProperty("difficulty")]
        public int Difficulty { get; set; }

        [JsonProperty("explanation")]
        public string Explanation { get; set; }

        public abstract Question Clone();
    }
}
